astropy/modeling/functional_models.py

```diff
@@ -2014,3 +2014,116 @@ def evaluate(cls, x, y, amplitude, r_eff, n, x_0, y_0, ellip, theta):
 
         return amplitude * np.exp(-bn * (z ** (1 / n) - 1))
+
+class CartesianShapelets(Fittable2DModel):
+    """
+    Two dimensional cartesian shapelets model.
+    Also computes Fourier Transform of shapelets.
+
+    Parameters
+    ----------
+    beta0 : float
+        Characteristic scale parameter for first
+        dimension basis functions.
+    beta1 : float
+        Characteristic scale parameter for second
+        dimension basis functions.
+    nmax : int
+        Maximum number of basis functions to use for each dimension.
+    phi : float, optional
+        Rotation angle in radians, counterclockwise from
+        the positive x-axis.
+    x_0 : float, optional
+        x position of the center.
+    y_0 : float, optional
+        y position of the center.
+
+    Notes
+    -----
+    Model formula:
+
+    .. math::
+
+        f(x) = \\sum_n f_n B_n (x; \\beta)
+
+    for :math:`f(x)` the flux of an object and
+    :math:`B_n (x; \\beta) = B_{n_1} (x_1; \\beta) B_{n_2} (x_2; \\beta)`,
+    where
+
+    .. math::
+    
+        B_n (x; \\beta) = \\left[2^n \\pi ^ {1/2} n! \\beta \\right]^{-1/2} H_n \\left( \\frac{x}{\\beta} \\right) e^{-\\frac{x^2}{2\\beta^2}}
+
+    (where :math:`H_n(x)` is a Hermite polynomial of order :math:`n`) and
+
+    .. math::
+
+        f_n = \\int_{-\\infty} ^{\\infty} d^2 x \\ f(x) B_n (x; \\beta)
+
+    Examples
+    --------
+    .. plot::
+        :include-source:
+
+        import numpy as np
+        from astropy.modeling.models import CartesianShapelets
+        import matplotlib.pyplot as plt
+
+        x,y = np.meshgrid(np.arange(31), np.arange(31))
+        mod = CartesianShapelets(beta0=2, beta1=2, nmax=6, phi=0,
+            x_0=15, y_0=15,)
+        img = mod(x, y) # returns gaussian by default
+
+        plt.figure()
+        plt.imshow(img, origin='lower', interpolation='nearest')
+        plt.xlabel('x')
+        plt.ylabel('y')
+        plt.show()
+
+    References
+    ----------
+    .. [1] Refregier, A. 2003, MNRAS 338, 35
+    .. [2] Refregier, A., & Bacon, D. 2003, MNRAS 338, 48
+    """
+
+    beta0 = Parameter(default=2.)
+    beta1 = Parameter(default=2.)
+    nmax = Parameter(default=6)
+    phi = Parameter(default=0)
+    x_0 = Parameter(default=0)
+    y_0 = Parameter(default=0)
+    _hermite = None
+
+    @classmethod
+    def evaluate(cls, x, y, beta0, beta1, nmax, phi, x_0, y_0, ft=False):
+        from scipy.misc import factorial
+        basis_func_arr = []
+        if cls._hermite == None:
+            try:
+                from scipy.special import hermite
+            except ValueError:
+                raise ImportError('CartesianShapelets model requires scipy > 0.11.')
+
+        for n0 in range(nmax):
+            for n1 in range(nmax):
+                h0 = hermite(n0)
+                h1 = hermite(n1)
+                rot_mat = np.array([[np.cos(phi),-1.*np.sin(phi)],[np.sin(phi),np.cos(phi)]]) # rotation matrix
```

Comment:
You don't need to create the `rot_mat` every iteration. Also `rotation_matrix` would be a better name.